### DISCUSSION TITLE
StatusEffects with targettype="Contained" sometimes cant access ItemComponents properties

### DISCUSSION THREAD

**Discussion by Ydrec** [FakeFishGames/Barotrauma#15309] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

StatusEffects with targettype="Contained" can not change properties of Components of Contained Items unless they are defined specifically inside \<Containable\>\</Containable\> element

### Reproduction steps

Here are 2 custom items to showcase this behaviour. Its a locker which is supposed to change `Flicker` property of a LightComponent of contained item when you interact with it. And an item which explodes if `Flicker` property is greater than 0.

<details>
intractable 
<summary>Example items</summary>

```xml
   <Items>
  <Item name="WIP_activecabinet" identifier="WIP_activecabinet" tags="container,suppliescontainer" linkable="true" pickdistance="150" waterproof="true" scale="0.5">
    <Sprite texture="Content/Items/Containers/containers.png" depth="0.84" sourcerect="889,349,120,132" origin="0.5,0.5" />
    <ItemContainer capacity="6" slotsperrow="3" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
        <GuiFrame relativesize="0.16,0.22" anchor="Center" style="ItemUI" />
        <Containable items="smallitem,mediumitem" excludedidentifiers="mobilecontainer" >
            <!-- This works fine and contained item explodes -->
            <!-- <StatusEffect type="OnContaining" target="Contained" interval="1" Flicker="1" TargetItemComponent="LightComponent"/>  -->
        </Containable>
    </ItemContainer>
    <Controller canbeselected="true" selectkey="Select" hidehud="false" drawuserbehind="false" noninteractablewhenflippedy="false" allowingameediting="false" drawhudwhenequipped="false" comparison="And">
        <!-- This doesnt work because statuseffect cant change properties of ItemComponents of Contained items when defined here-->
        <StatusEffect type="OnActive" target="Contained" interval="1" Flicker="1" TargetItemComponent="LightComponent"/>

        <!-- When targeting NearbyItems instead statuseffect can access ItemComponent properties  -->
        <!-- <StatusEffect type="OnActive" target="NearbyItems" Range="1" interval="1" Flicker="1" TargetItemComponent="LightComponent"/> -->
  </Controller>
  </Item>

  <Item name="WIP_expoldingitem" identifier="WIP_expoldingitem" category="Material" maxstacksize="32" maxstacksizecharacterinventory="8" Tags="smallitem" canbepicked="true" description="" scale="0.5" >
    <InventoryIcon texture="Content/Items/InventoryIconAtlas.png" sourcerect="64,192,64,64" />
    <Sprite texture="Content/Items/Materials/minerals.png" sourcerect="29,7,23,44" depth="0.55" origin="0.5,0.5" />
    <Body width="22" height="42" density="20" />
    <Holdable canBeCombined="true" removeOnCombined="true" slots="Any,RightHand,LeftHand" handle1="0,0" msg="ItemMsgPickUpSelect" />
    <LightComponent LightColor="255,255,255,255" Flicker="0" range="100">
      <StatusEffect type="Always" target="This">
        <Conditional Flicker="gt 0" TargetItemComponent="LightComponent"/>
        <Explosion range="150.0" force="1" camerashake="20.0" shockwave="true" smoke="false" flames="false" sparks="false" underwaterbubble="false"/>
        <Remove />
      </StatusEffect>
    </LightComponent>
  </Item>
</Items>
```

</details>

When you place exploding item inside cabinet it should almost immediately explode and delete itself. The 2 commented out sections work as expected but not StatusEffect with target="Containable" in Controller 

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by PanmanS** [comment]
This is real and true

**Comment by Ydrec** [comment]
That same statuseffect can change properties of the Item itself, like condition so it only cant access Component properties.

**Comment by zhu-rengong** [comment]
Because the statuseffect only considers the item itself, not its components:
https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaShared/SharedSource/Items/Item.cs#L2021

I guess that if this "bug" is fixed, the whole statuseffect system will consume a lot of computing resources in Vanilla.

**Reply by Ydrec** [reply]
It already considers components in 2 other situations shown in example. Seems like inconsistent behavior to me.

**Comment by NilanthAnimosus** [comment]
Closing as merged into the codebase. Should be in a future build.
